lib/services/platform-service.ts

```diff
@@ -177,13 +177,62 @@ export class PlatformService implements IPlatformService {
 			platform = platform.toLowerCase();
 
 			this.preparePlatform(platform).wait();
-			if(options.emulator) {
+            if (options.emulator) {
 				this.deployOnEmulator(platform).wait();
 			} else {
 				this.deployOnDevice(platform).wait();
 			}
 		}).future<void>()();
-	}
+    }
+
+    public debugPlatform(platform: string): IFuture<void> {
+        return (() => {
```

Comment:
You can remove starting new fiber and directly return the results of called functions.